Updates to travel/history app needles

pwithnall opened the pull request:
Following up from the errors found in [this test run](https://openqa.dev.endlessm.com/tests/overview?distri=eos&version=master&build=181207-002305&groupid=1). See also [this task](https://phabricator.endlessm.com/T24336).

https://phabricator.endlessm.com/T24855

(Still working through the sequence to make sure the tests pass in their entirety, so this isn’t quite ready for review yet.)

wjt commented:
>  needles: Update some outdated history screenshots 

The new copyright screenshot seems wrong. Surely there are still photographs in the history app which we did not take?

wjt commented on tests/desktop_history.pm:
```diff
@@ -11,30 +11,41 @@ sub run {
+
+    # Enter into a random category. (The test was originally designed to enter
+    # the ‘Old Age’ category, but since the category tiles rotate regularly we
+    # can’t guarantee that.)
```
Is there some way we can set a fixed seed so that the tiles really are always the same? eg. is it derived from the current date? Could the SDK provide a way to do this?

wjt replied:
(Doesn't deal with content updates, of course.)

pwithnall replied:
I think that’s something to deal with in a follow-up issue. Filed https://phabricator.endlessm.com/T25287.

wjt commented on tests/desktop_dinosaurs.pm:
```diff
@@ -8,8 +8,7 @@ sub run {
     check_desktop_clean();
-    $self->add_nightly_sdk_repo();
-    $self->install_app($app_id, run_with_nightly_sdk => 1);
+    $self->install_app($app_id);
```
This is only true, I guess, because right now the nightly and released SDKs are very close together. If/when development begins again on the SDK, it would seem sensible to test both cases.

pwithnall replied:
I’m fairly against doing that. We should be testing the images as they are generated; that’s the whole point of OpenQA. Why do we not use the nightly SDK by default on our nightly EOS master images?

pwithnall commented:
> The new copyright screenshot seems wrong. Surely there are still photographs in the history app which we did not take?

Indeed. I filed a few bugs, and will update my commit to include a `FIXME` comment linking them:
 * https://phabricator.endlessm.com/T25284
 * https://phabricator.endlessm.com/T25285
 * https://phabricator.endlessm.com/T25286

pwithnall commented:
> Is there some way we can mark tests "xfail"?

Yes, good suggestion. I’ve pushed a commit which does this for the travel test (and added a link to the Phab task to the rest of the commit messages; it now looks like I’ve rebased and made lots of changes, but pushing this xfail commit is actually the only one).

wjt commented:
Looks fine, could you trigger a run?

pwithnall commented:
* History, passes entirely: https://openqa.endlessm.com/tests/14720
 * Travel, fails as expected, but that’s treated as an xfail: https://openqa.endlessm.com/tests/14721#

Merging accordingly, thanks!
